pkg/agent/endpoints/sdsv3/handler.go

```diff
@@ -507,14 +508,38 @@ func supportsSPIFFEAuthExtension(req *discovery_v3.DiscoveryRequest) bool {
 }
 
 func (h *Handler) isSPIFFECertValidationDisabled(req *discovery_v3.DiscoveryRequest) bool {
-	if h.c.DisableSPIFFECertValidation {
+	fields := req.Node.GetMetadata().GetFields()
+	v, ok := fields[disableSPIFFECertValidationKey]
+	var vb bool
+	if ok {
+		var err error
+		vb, err = parseBool(v)
+		if err != nil {
+			// error means that value isn't parsable string
+			// so it would be safer to assume that key doesn't exist in envoy node metadata
+			ok = false
+		}
+	}
+
+	if h.c.DisableSPIFFECertValidation && (!ok || vb) {
```

Comment:
nit: this totally works but I think it could be a little more readable if things were reordered a little:
```suggestion
	disabled := h.c.DisableSPIFFECertValidation
	if v, ok := req.Node.GetMetadata().GetFields()[disableSPIFFECertValidationKey]; ok {
		if override, err := parseBool(v); err == nil {
			disabled = override
		}
	}
	return disabled
```

Reply:
agree, changed it